plugins-compat-tester/src/main/java/org/jenkins/tools/test/hook/MultiParentCompileHook.java

```diff
@@ -113,12 +119,60 @@ private void copy(Path eslintrc, File pluginFolder) {
         return mconfig;
     }
 
-    private void compile(MavenRunner.Config mavenConfig, File path) throws PomExecutionException, IOException {
+    private void compile(MavenRunner.Config mavenConfig, File path, File localCheckoutDir, String parentFolder, String pluginName) throws PomExecutionException, IOException {
+        if (isSnapshotMultiParentPlugin(parentFolder, path, localCheckoutDir)) {
+            // "process-test-classes" not working properly on multi-module plugin. See https://issues.jenkins.io/browse/JENKINS-62658
+            // using "mvn clean install -DskipTests -Dinvoker.skip -Denforcer.skip to skip testing and just compile and install dependencies"
+            String mavenModule = getMavenModule(pluginName, path, runner, mavenConfig);
+            if (StringUtils.isBlank(mavenModule)) {
+                throw new IOException(String.format("Unable to retrieve the Maven module for plugin %s on %s", pluginName, path));
+            }
+            runner.run(mavenConfig, path.getParentFile(), setupCompileResources(path.getParentFile()), "clean", "install", "-DskipTests", "-Dinvoker.skip", "-Denforcer.skip", "-Dmaven.javadoc.skip", "-am", "-pl", mavenModule);
+        } else {
+            runner.run(mavenConfig, path, setupCompileResources(path), "clean", "process-test-classes", "-Dmaven.javadoc.skip");
+        }
+    }
+
+    /**
+     * Checks if a plugin is a multiparent plugin with a SNAPSHOT project.version and 
+     * without local checkout directory overriden. 
+     * 
+     * @param parentFolder
+     * @param path
+     * @param localCheckoutDir
+     * @return
+     * @throws PomExecutionException 
+     * @throws IOException 
+     */
+    private boolean isSnapshotMultiParentPlugin(String parentFolder, File path, File localCheckoutDir) throws PomExecutionException, IOException {
+        if (localCheckoutDir != null) {
+            return false;
+        }
```

Comment:
I do not understand why this clause is here. As discussed earlier, having a local checkout dir (with a plain old MRP plugin) is one of the reasons why you _might_ in fact need this patch.

Reply:
:thinking: you are right, I will review this in depth. I included at the very beginning because I wrongly understood https://github.com/jenkinsci/plugin-compat-tester/pull/318#discussion_r752494751

Reply:
Definitely an edge case, not very important, just a point of confusion.